src/middlewared/middlewared/alert/source/volume_version.py

```diff
@@ -10,45 +9,22 @@ class VolumeVersionAlertClass(AlertClass):
     level = AlertLevel.WARNING
     title = "New Feature Flags Are Available for Pool"
     text = (
-        "New feature flags are available for volume %s. Refer "
-        "to the \"Upgrading a ZFS Pool\" subsection in the "
-        "User Guide \"Installing and Upgrading\" chapter "
-        "and \"Upgrading\" section for more instructions."
-    )
-
-
-class ZfsVersionOutOfDateAlertClass(AlertClass):
-    category = AlertCategory.STORAGE
-    level = AlertLevel.WARNING
-    title = "ZFS Filesystem Version Is Out of Date"
-    text = (
-        "ZFS filesystem version is out of date. Please consider upgrading it. See <a href=\""
-        "https://www.ixsystems.com/documentation/freenas/11.2/install.html#upgrading-a-zfs-pool\">"
+        "New ZFS version or feature flags are available for pool %s. Please see <a href=\""
+        "https://www.truenas.com/docs/hub/tasks/advanced/upgrading-pool/\">"
         "Upgrading a ZFS Pool</a> for details."
     )
 
 
-class VolumeVersionAlertSource(ThreadedAlertSource):
+class VolumeVersionAlertSource(AlertSource):
     schedule = IntervalSchedule(timedelta(minutes=5))
 
-    def check_sync(self):
+    async def check(self):
         alerts = []
-        for pool in self.middleware.call_sync("pool.query"):
-            if not self.middleware.call_sync('pool.is_upgraded', pool["id"]):
+        for pool in await self.middleware.call("pool.query"):
+            if not await self.middleware.call("pool.is_upgraded", pool["id"]):
                 alerts.append(Alert(
                     VolumeVersionAlertClass,
                     pool["name"],
                 ))
 
-        proc = subprocess.Popen(
-            "zfs upgrade | grep FILESYSTEM",
-            shell=True,
-            stdout=subprocess.PIPE,
-            stderr=subprocess.PIPE,
-            encoding="utf8",
-        )
-        output = proc.communicate()[0].strip(" ").strip("\n")
-        if output:
-            alerts.append(Alert(ZfsVersionOutOfDateAlertClass))
```

Comment:
Why was this removed? I think zfs upgrade is different than zpool upgrade?

Reply:
Because ZFS version 4 is something you won't find these days, and our docs don't even mention how to upgrade ZFS (the link was pointing to zpool upgrade page)